### PR TITLE
Populate metric selector for custom metrics

### PR DESCRIPTION
**Pull request by dgrisonnet**
When querying custom-metrics, the metric label selectors weren't populated in the resulting values.

A query with a metricLabelSelector query parameter set to `endpoint=https` was resulting in the following MetricValueList:
```
{
  "kind": "MetricValueList",
  "apiVersion": "custom.metrics.k8s.io/v1beta1",
  ...
  "items": [
    {
      ...
      "selector": null
    }
    ...
}
```

With this change, it now appears as:
```
{
  "kind": "MetricValueList",
  "apiVersion": "custom.metrics.k8s.io/v1beta1",
  ...
  "items": [
    {
      ...
      "selector": {
        "matchLabels": {
          "endpoint": "https"
        }
      }
    }
    ...
}
```

/cc @s-urbaniak

### CODE REVIEW

File: pkg/custom-provider/provider.go [hunk 1]
```diff
@@ -90,18 +90,29 @@ func (p *prometheusProvider) metricFor(value pmodel.SampleValue, name types.Name
...
+			return nil, err
+		}
+		metric.Metric.Selector = sel
+	}
+
+	return metric, nil
 }
 
-func (p *prometheusProvider) metricsFor(valueSet pmodel.Vector, info provider.CustomMetricInfo, namespace string, names []string) (*custom_metrics.MetricValueList, error) {
+func (p *prometheusProvider) metricsFor(valueSet pmodel.Vector, namespace string, names []string, info provider.CustomMetricInfo, metricSelector labels.Selector) (*custom_metrics.MetricValueList, error) {
```
Comment (dgrisonnet) [inline pkg/custom-provider/provider.go:115]: note: I changed the order of the parameters here to match the `metricFor` function

**Comment by s-urbaniak** [comment]
lgtm, thank you!

**Comment by s-urbaniak** [comment]
@dgrisonnet just because i simply forgot: as a follow-up we could potentially add some tests, wdyt?

**Comment by dgrisonnet** [comment]
yes, that would definitely be a great addition. However, since I am not too familiar with prometheus-adapter test framework, it might take me some times.

**Pull request merged**